### PR TITLE
Homework 9

### PR DESCRIPTION
**Pull request by DimitryusX**

### CODE REVIEW

File: Homework9/Utils.cpp [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        else if (numbers[i] < 0)
+        {
+            numbers[i] = 0;
+        }
+    }
+
+    for (int i = 0; i < size; i++)
+    {
+        std::cout << numbers[i] << " ";
+    }
```
Comment (helen-black) [inline Homework9/Utils.cpp:47]: functions shouldn't speak

File: Homework9/Utils.cpp [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    {
+        if (str[i] >= 'a' && str[i] <= 'z')
+        {
+            std::cout << static_cast<char>(str[i] - 32);
+        }
+        else
+        {
+            std::cout << str[i];
+        }
+    }
```
Comment (helen-black) [inline Homework9/Utils.cpp:62]: don't print here, please. just update input str

File: Homework9/Utils.cpp [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+    do {
+        if (word[i] != reversed[i]) {
+            return false;
+        }
+        i++;
+    } while (word[i] != '\0');
+
+    return true;
+}
```
Comment (helen-black) [inline Homework9/Utils.cpp:86]: there is easuer way to solve this task

File: Homework9/Utils.cpp [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+                    vowelsCount++;
+                    break;
+                default:
+                    consonantsCount++;
+            }
+        }
+    }
+}
+
+bool isEqual(const char str1[], const char str2[])
```
Comment (helen-black) [inline Homework9/Utils.cpp:109]: test with 'abs' and 'absd'

**Review by helen-black — CHANGES_REQUESTED** [review]

File: Homework9/Utils.cpp (hunk 3 above)
Comment (DimitryusX) [reply Homework9/Utils.cpp]: Fixed.

File: Homework9/Utils.cpp (hunk 4 above)
Comment (DimitryusX) [reply Homework9/Utils.cpp:90]: Fixed.

File: Homework9/Utils.cpp (hunk 2 above)
Comment (DimitryusX) [reply Homework9/Utils.cpp]: Fixed.

File: Homework9/Utils.cpp (hunk 1 above)
Comment (DimitryusX) [reply Homework9/Utils.cpp]: Fixed.

File: Homework9/Utils.cpp [hunk 5]
```diff
@@ -68,21 +59,11 @@ bool isPalindrom(std::string word)
 
     int i = 0;
 
-    do {
+    for (int i = word.length() - 1; i >= 0; --i) {
         reversed += word[i];
```
Comment (helen-black) [inline Homework9/Utils.cpp:63]: it is the same algorithm

File: Homework9/Utils.cpp [hunk 6]
```diff
@@ -108,7 +89,11 @@ void parseStringLetters(const char str[], int& vowelsCount, int& consonantsCount
 
 bool isEqual(const char str1[], const char str2[]) 
 {
-    for (int i = 0; str1[i] != '\0'; i++) {
+    for (int i = 0; str1[i] != '\0' || str2[i] != '\0'; i++) {
+
+        char test1 = str1[i];
+        char test2 = str2[i];
+
```
Comment (helen-black) [inline Homework9/Utils.cpp:96]: you don't use it

**Review by helen-black — CHANGES_REQUESTED** [review]

File: Homework9/Utils.cpp (hunk 6 above)
Comment (DimitryusX) [reply Homework9/Utils.cpp:96]: @helen-black yes, my mistake, forgot to remove debug.

File: Homework9/Utils.cpp (hunk 5 above)
Comment (DimitryusX) [reply Homework9/Utils.cpp:63]: @helen-black can't find a better solution. :(

Comment (DimitryusX) [reply Homework9/Utils.cpp:63]: @helen-black I can compare the original word with the reverted, don't understand how it is possible to add more optimization.

**Pull request merged**